### PR TITLE
Redesign2

### PR DESCRIPTION
**Pull request by exacs**
Redesign and re-implementation of SIS Monitor. I have created a new PR because this one has a different design after a discussion we had on the strategy to monitor SIS Import Errors.

---

## Description of the strategy:

1. Every app that send files to Canvas will send them with the app name as filename prefix. For example, see https://github.com/KTH/lms-activity-rooms/pull/18
2. For now, this LMS SIS Monitor will only look at errors that happen on `lms-activity-rooms` because errors that came from that application will almost never be false positives.
3. You could test this PR before the other one is merged. That will mean **LMS SIS Monitor doesn't catch any errors yet but is ready to start catching them.**
4. Because of point number 2, I have decided to remove all checks of false-positive errors. It might come back in a future.

---

- The important file is [`src/log-errors.js`](https://github.com/KTH/lms-sis-monitor/blob/redesign2/src/log-errors.js) which contains the logic of monitoring errors
- This PR also includes other minor fixes like filename changes.

---

## 👀 Tips for testing this application

This application runs a periodical task. However, it is separated the "run a task periodically" and the "task itself" since the latter is more important to test.

There is a file called [`scripts/errors-last-hour`](https://github.com/KTH/lms-sis-monitor/blob/redesign2/scripts/errors-last-hour.js) that runs the task without the need of running the server.

### CODE REVIEW

File: src/log-errors.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  log.info(`Fetching from ${startDate.toISOString()}`)
+  const imports = canvas.listPaginated('/accounts/1/sis_imports', {
+    created_since: startDate.toISOString()
+  })
+
+  for await (const page of imports) {
+    for (const imp of page.sis_imports) {
+      const attachments = imp.csv_attachments
+
+      if (imp.workflow_state !== 'imported') {
```
Comment (me97esn) [inline src/log-errors.js:48]: Can you add a comment here please? Why are these two if:s needed?

**Review by me97esn — CHANGES_REQUESTED** [review]
Just a minor change requested

File: app.js [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
-skog.createLogger({
+require('skog/bunyan').createLogger({
   app: 'lms-sis-monitor',
   name: 'lms-sis-monitor',
   level: 'info'
 })
 
-cron.start()
+const server = require('./src/server')
+const runPeriodically = require('./run-periodically')
```
Comment (me97esn) [inline app.js:10]: Shouldn't this be ./src/run-periodically?

**Review by me97esn — CHANGES_REQUESTED** [review]
Have you tested this PR @exacs ? It crashes when I try to run it locally, because it tries to find ./run-periodically when it's actually in the `src` folder...
This is why I don't want to use a script like `errors-last-hour.js` when I test the code, but rather think you should run the actual app when testing it

File: src/log-errors.js (hunk 1 above)
Comment (exacs) [reply src/log-errors.js:51]: I thought to "return everything if `prefixes` is an empty array" but it's strange behavior actually.

Answer: the two ifs are not needed :)

**Review by exacs — COMMENTED** [review]
Good catch of the error @me97esn!! I'll keep in mind your comment! I think it is ok to have the `errors-last-hour` script anyways _after_ we are sure that the app runs

**Comment by me97esn** [comment]
When testing this I think we should make a minor change that makes it easier to debug the errors:
Since we might log the same error multiple times (for instance if we see an error, fix the underlying issue and then restart the lms-sis-monitor app), I think it would be really valuable information to log the _time_ of the sis import. That could make it obvious if this is a new error or something we already have fixed

**Review by me97esn — CHANGES_REQUESTED** [review]
See my comment under conversations: I would like us to log the time of the sis import in the error message

**Pull request merged**